tests/models/mamba/test_modeling_mamba.py

```diff
@@ -197,7 +197,25 @@ def create_and_check_state_equivalency(self, config, input_ids, *args):
         self.parent.assertTrue(torch.allclose(torch.cat([output_one, output_two], dim=1), output_whole, atol=1e-5))
         # TODO the orignal mamba does not support decoding more than 1 token neither do we
 
-    def create_and_check_forward_and_backwards(self, config, input_ids, *args, gradient_checkpointing=False):
+    def create_and_check_mamba_cached_slow_forward_and_backwards(
+        self, config, input_ids, *args, gradient_checkpointing=False
+    ):
+        model = MambaModel(config)
+        model.to(torch_device)
+        if gradient_checkpointing:
+            model.gradient_checkpointing_enable()
+
+        # creating and using cache
+        _ = model(input_ids[:, :-1], use_cache=True)
+        outputs = model._slow_forward(input_ids[:, :-1], use_cache=True).last_hidden_state
```

Comment:
I believe the method for the model isn't private. 

```suggestion
        outputs = model.slow_forward(input_ids[:, :-1], use_cache=True).last_hidden_state
```

t's funny this wasn't picked up in the tests. Can you confirm that you can run these tests and everything passes (i.e. they're not skipped)

Reply:
I noticed that I used the torch module's `_slow_forward`, hence why it's "private". Guess I was doing this too hastily but the new commit should really use the mamba mixer `slow_forward` now. Hope, that makes it clearer. 

And here's a google colab [notebook](https://colab.research.google.com/drive/1kH6bqzylGSupJWBTA4qBxPCSl19c0lGJ?usp=sharing) showing that the generated tensors are indeed identical.